packages/vike-node/src/plugin/plugins/devServerPlugin.ts

```diff
@@ -114,13 +114,18 @@ export function devServerPlugin(): Plugin {
     // biome-ignore lint/suspicious/noExplicitAny: <explanation>
     const server = (req.socket as any).server as Server
     server.on('upgrade', (clientReq, clientSocket, wsHead) => {
-      if (clientReq.url === VITE_HMR_PATH) {
+      if (isHMRProxyRequest(clientReq)) {
         assert(HMRServer)
         HMRServer.emit('upgrade', clientReq, clientSocket, wsHead)
       }
     })
     // true if we need to send an empty Response waiting for the upgrade
-    return req.url === VITE_HMR_PATH
+    return isHMRProxyRequest(req)
+  }
+
+  function isHMRProxyRequest(req: IncomingMessage) {
+    const url = new URL(`http://example.com${req.url}`)
```

Comment:
URL constructor have a native (more robust) way to handle URL base. See https://developer.mozilla.org/en-US/docs/Web/API/URL/URL#parameters

```suggestion
    const url = new URL(req.url, "http://example.com")
```

Reply:
Check out [this commit](https://github.com/vikejs/vike-node/pull/52/commits/70e86f9902df7d4b4c542af86ac4696e02ae09e1) which switched to use the two argument constructor as you suggested.
The case where `req.url === undefined` is handled explicitly.
Semantics remain the same if one assumes `VITE_HMR_PATH` to always be a path starting with `/`.